From a452d8e1cfb7aa8f5b8774a7f06959a12cf7af24 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Mon, 28 Apr 2025 12:02:11 +0200
Subject: [PATCH] dt-bindings: rtc: stm32: fix dt_binding_check errors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Unquote $ref value.
Fix allOf if instruction using 'false' value for properties not required
at alli.

Change-Id: I25e7a4c65369ee16d6bdc3fb3d30323b473a9455
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/455180
Tested-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 .../devicetree/bindings/rtc/st,stm32-rtc.yaml    | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml b/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
index 9e86c4cd4548..8d67c5f34fc6 100644
--- a/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
+++ b/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
@@ -54,14 +54,14 @@ properties:
     maxItems: 1
 
   st,lsco:
-    $ref: "/schemas/types.yaml#/definitions/uint32"
+    $ref: /schemas/types.yaml#/definitions/uint32
     description: |
       To select and enable RTC Low Speed Clock Output.
       Refer to <include/dt-bindings/rtc/rtc-stm32.h> for the supported values.
       Pinctrl state named "default" may be defined to reserve pin for RTC output.
 
   st,alarm:
-    $ref: "/schemas/types.yaml#/definitions/uint32"
+    $ref: /schemas/types.yaml#/definitions/uint32
     description: |
       To select and enable RTC Alarm A output.
       Refer to <include/dt-bindings/rtc/rtc-stm32.h> for the supported values.
@@ -80,11 +80,9 @@ allOf:
           minItems: 1
           maxItems: 1
 
-        st,lsco:
-          maxItems: 0
+        st,lsco: false
 
-        st,alarm:
-          maxItems: 0
+        st,alarm: false
 
         clock-names: false
 
@@ -103,11 +101,9 @@ allOf:
           minItems: 2
           maxItems: 2
 
-        st,lsco:
-          maxItems: 0
+        st,lsco: false
 
-        st,alarm:
-          maxItems: 0
+        st,alarm: false
 
       required:
         - clock-names
-- 
2.50.1

